core/src/main/java/org/bitcoinj/core/TransactionOutPoint.java

```diff
@@ -85,28 +97,40 @@ public TransactionOutPoint(TransactionOutput connectedOutput) {
     }
 
     /**
-     * Deserializes the message. This is usually part of a transaction message.
-     * @throws ProtocolException
+     * Write this transaction outpoint into the given buffer.
+     *
+     * @param buf buffer to write into
+     * @return the buffer
+     * @throws BufferOverflowException if the outpoint doesn't fit the remaining buffer
      */
-    public TransactionOutPoint(ByteBuffer payload) throws ProtocolException {
-        super(payload);
+    public ByteBuffer write(ByteBuffer buf) throws BufferOverflowException {
+        buf.put(hash.getReversedBytes());
+        ByteUtils.writeInt32LE(index, buf);
+        return buf;
     }
 
-    @Override
-    protected void parse(ByteBuffer payload) throws BufferUnderflowException, ProtocolException {
-        hash = Sha256Hash.read(payload);
-        index = ByteUtils.readUint32(payload);
+    /**
+     * Allocates a buffer and writes this transaction outpoint into it. The buffer is backed by a byte array and it's
+     * already rewound for reading.
+     *
+     * @return buffer containing the transaction outpoint
+     */
+    public ByteBuffer buffer() {
+        ByteBuffer buf = write(ByteBuffer.allocate(BYTES));
+        ((Buffer) buf).rewind();
+        return buf;
```

Comment:
Maybe we should just have a `bytes()` method that returns an array. It seems every usage is `.buffer().array()` so it might be better to skip the intermediate `ByteBuffer`?

Reply:
I see what you mean, the ideal form isn't found yet. (There is precident in `Services`.)

Reply:
Ok, it's now `byte[] serialize()`.